google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py

```diff
@@ -200,15 +200,18 @@ def _process_requests(
                 exc = AcknowledgeError(AcknowledgeStatus.FAILED_PRECONDITION, info=None)
             else:
                 exc = AcknowledgeError(AcknowledgeStatus.OTHER, str(error_status))
-            future = ack_reqs_dict[ack_id].future
-            future.set_exception(exc)
-            requests_completed.append(ack_reqs_dict[ack_id])
+            if ack_id in ack_reqs_dict:
+                future = ack_reqs_dict[ack_id].future
+                if future is not None:
+                    future.set_exception(exc)
+                    requests_completed.append(ack_reqs_dict[ack_id])
         # Since no error occurred, requests with futures are completed successfully.
         elif ack_reqs_dict[ack_id].future:
             future = ack_reqs_dict[ack_id].future
             # success
-            future.set_result(AcknowledgeStatus.SUCCESS)
-            requests_completed.append(ack_reqs_dict[ack_id])
+            if future is not None:
+                future.set_result(AcknowledgeStatus.SUCCESS)
+                requests_completed.append(ack_reqs_dict[ack_id])
```

Comment:
Unindent `requests_completed.append(ack_reqs_dict[ack_id])` to ensure the request is completed.